packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -149,6 +149,12 @@ abstract class GlobalKey<T extends State<StatefulWidget>> extends Key {
     assert(() {
       assert(parent != null);
       if (_debugReservations.containsKey(this) && _debugReservations[this] != parent) {
+        if (_debugReservations[this].renderObject != null) {
```

Comment:
It's probably worth adding a very brief comment here (like below) explaining why this may happen.

Reply:
added